catalog/schematron/catalog-schematron-plugin/src/main/java/ddf/services/schematron/SchematronValidationService.java

```diff
@@ -73,538 +71,198 @@
-        String entryDocument = catalogEntry.getMetadata();
-        LOGGER.debug("entryDocument: {}", entryDocument);
-
-        // Create a Reader for the catalog entry's contents
-        StringReader entryDocumentReader = new StringReader(entryDocument);
+    public void setSuppressWarnings(boolean suppressWarnings) {
+        this.suppressWarnings = suppressWarnings;
+    }
 
-        try {
-            // Using the precompiled/stored Schematron validator, validate the catalog entry's
-            // contents
-            Transformer transformer = validator.newTransformer();
-            DOMResult schematronResult = new DOMResult();
-            transformer.transform(new StreamSource(entryDocumentReader), schematronResult);
-            this.report = new SvrlReport(schematronResult);
-
-            LOGGER.trace("SVRL Report:\n\n{}", report.getReportAsText());
-
-            // If the Schematron validation failed, then throw an exception with details of the
-            // errors
-            // and warnings from the Schematron report included in the exception that is thrown to
-            // the client.
-            if (!this.report.isValid(this.suppressWarnings)) {
-                StringBuffer errorMessage = new StringBuffer(
-                        "Schematron validation failed for catalog entry #" + catalogEntryNum
-                                + ".\n\n");
-                List<String> errors = this.report.getErrors();
-                LOGGER.debug("errors.size() = {}", errors.size());
-                for (String error : errors) {
-                    errorMessage.append(error);
-                    errorMessage.append("\n");
-                }
-
-                // If warnings are to be included from the Schematron report as part of the errors
-                // message
-                if (!this.suppressWarnings) {
-                    List<String> warnings = this.report.getWarnings();
-                    LOGGER.debug("warnings.size() = {}", warnings.size());
-                    for (String warning : warnings) {
-                        LOGGER.debug("warning = {}", warning);
-                        errorMessage.append(warning);
-                        errorMessage.append("\n");
-                    }
-
-                }
-                throw new StopProcessingException(errorMessage.toString());
+    public void setSchematronFileNames(List<String> schematronFileNames) {
+        this.schematronFileNames = new ArrayList<>();
+        for (String filename : schematronFileNames) {
+            String fullpath = Paths.get(filename).toString();
+            if (!Paths.get(filename).isAbsolute()) {
+                fullpath = Paths.get(SCHEMATRON_BASE_FOLDER, fullpath).toString();
             }
+            this.schematronFileNames.add(fullpath);
+        }
+        try {
+            updateValidators();
+        } catch (SchematronInitializationException e) {
+            // silently ignore
```

Comment:
Why not log something here? I imagine silently failing would be a lot harder to debug